Compile on mono error

With latest master (without PCL):

```
$ ./build.sh
/Library/Frameworks/Mono.framework/Versions/4.0.0/lib/mono/xbuild/12.0/bin/Microsoft.Common.targets:  warning : Reference 'PresentationCore' not resolved
/Library/Frameworks/Mono.framework/Versions/4.0.0/lib/mono/xbuild/12.0/bin/Microsoft.Common.targets:  warning : Reference 'PresentationFramework' not resolved
ItemControl.xaml.cs(3,22): error CS0234: The type or namespace name `Controls' does not exist in the namespace `System.Windows'. Are you missing an assembly reference?
ItemControl.xaml.cs(6,40): error CS0246: The type or namespace name `UserControl' could not be found. Are you missing an assembly reference?
MainWindow.xaml.cs(6,39): error CS0246: The type or namespace name `Window' could not be found. Are you missing an assembly reference?
```

This is due to WPF not existing on mono. Removing the ToDo* projecs from the solution solves the problem.

--- a/Benchmarks/AsyncOverhead/AsyncOverhead.fsproj
+++ b/Benchmarks/AsyncOverhead/AsyncOverhead.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -43,6 +42,7 @@
     </WarningsAsErrors>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -72,7 +72,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/Cell/Cell.fsproj
+++ b/Benchmarks/Cell/Cell.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -40,6 +39,7 @@
     <DocumentationFile>bin\Release\Cell.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -73,7 +73,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/Chameneos/Chameneos.fsproj
+++ b/Benchmarks/Chameneos/Chameneos.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -45,6 +44,7 @@
     </NoWarn>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -74,7 +74,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/CmlLCH/CmlLCH.fsproj
+++ b/Benchmarks/CmlLCH/CmlLCH.fsproj
@@ -19,7 +19,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -41,6 +40,7 @@
     <DocumentationFile>bin\Release\CmlLCH.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -86,7 +86,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/CounterActor/CounterActor.fsproj
+++ b/Benchmarks/CounterActor/CounterActor.fsproj
@@ -20,7 +20,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -38,6 +37,9 @@
     <DocumentationFile>bin\Release\CounterActor.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
+    <DefineConstants>
+    </DefineConstants>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -67,7 +69,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/Fibonacci/Fibonacci.fsproj
+++ b/Benchmarks/Fibonacci/Fibonacci.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -47,6 +46,7 @@
     <OtherFlags>--warnon:1182</OtherFlags>
     <NoWarn>0075</NoWarn>
     <Prefer32Bit>false</Prefer32Bit>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -76,7 +76,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/PingPong/PingPong.fsproj
+++ b/Benchmarks/PingPong/PingPong.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -38,6 +37,9 @@
     <DocumentationFile>bin\Release\PingPong.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
+    <DefineConstants>
+    </DefineConstants>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -67,7 +69,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/PostMailbox/PostMailbox.fsproj
+++ b/Benchmarks/PostMailbox/PostMailbox.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -40,6 +39,7 @@
     <DocumentationFile>bin\Release\PostMailbox.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -83,7 +83,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/PrimesStream/PrimesStream.fsproj
+++ b/Benchmarks/PrimesStream/PrimesStream.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -44,6 +43,7 @@
     <NoWarn>0075</NoWarn>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -73,7 +73,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/ReaderWriter/ReaderWriter.fsproj
+++ b/Benchmarks/ReaderWriter/ReaderWriter.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -39,6 +38,7 @@
     <DocumentationFile>bin\Release\ReaderWriter.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -68,7 +68,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/StartRing/StartRing.fsproj
+++ b/Benchmarks/StartRing/StartRing.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>false</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -40,6 +39,7 @@
     <DocumentationFile>bin\Release\StartRing.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -73,7 +73,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Benchmarks/Streams/Streams.fsproj
+++ b/Benchmarks/Streams/Streams.fsproj
@@ -16,7 +16,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug</OutputPath>
     <DefineConstants>DEBUG</DefineConstants>
@@ -36,6 +35,8 @@
     <Externalconsole>true</Externalconsole>
     <Tailcalls>true</Tailcalls>
     <Prefer32Bit>false</Prefer32Bit>
+    <DefineConstants>
+    </DefineConstants>
   </PropertyGroup>
   <Choose>
     <When Condition="'$(VisualStudioVersion)' == '11.0'">
@@ -65,7 +66,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />
@@ -73,34 +73,27 @@
     <Reference Include="System.Numerics" />
     <ProjectReference Include="..\..\Libs\Hopac.Core\Hopac.Core.csproj">
       <Name>Hopac.Core</Name>
-      <Project>{92ac6198-fa63-4458-afec-2bfbb2b0d914}</Project>
-      <Private>True</Private>
+      <Project>{92AC6198-FA63-4458-AFEC-2BFBB2B0D914}</Project>
     </ProjectReference>
     <ProjectReference Include="..\..\Libs\Hopac.Platform.Net\Hopac.Platform.Net.fsproj">
       <Name>Hopac.Platform.Net</Name>
-      <Project>{b285aeb6-fc96-4212-b7a2-6b73597ce607}</Project>
-      <Private>True</Private>
+      <Project>{B285AEB6-FC96-4212-B7A2-6B73597CE607}</Project>
     </ProjectReference>
     <ProjectReference Include="..\..\Libs\Hopac\Hopac.fsproj">
       <Name>Hopac</Name>
-      <Project>{898d8fe3-c9be-4115-9a13-7615af27c048}</Project>
-      <Private>True</Private>
+      <Project>{898D8FE3-C9BE-4115-9A13-7615AF27C048}</Project>
     </ProjectReference>
     <Reference Include="System.Reactive.Core">
       <HintPath>..\..\packages\Rx-Core.2.2.5\lib\net45\System.Reactive.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="System.Reactive.Interfaces">
       <HintPath>..\..\packages\Rx-Interfaces.2.2.5\lib\net45\System.Reactive.Interfaces.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="System.Reactive.Linq">
       <HintPath>..\..\packages\Rx-Linq.2.2.5\lib\net45\System.Reactive.Linq.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="System.Reactive.PlatformServices">
       <HintPath>..\..\packages\Rx-PlatformServices.2.2.5\lib\net45\System.Reactive.PlatformServices.dll</HintPath>
-      <Private>True</Private>
     </Reference>
   </ItemGroup>
 </Project>

--- a/Benchmarks/ThreadRing/ThreadRing.fsproj
+++ b/Benchmarks/ThreadRing/ThreadRing.fsproj
@@ -20,7 +20,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -36,6 +35,9 @@
     <DocumentationFile>bin\Release\ThreadRingCh.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
+    <DefineConstants>
+    </DefineConstants>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -57,7 +59,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Examples/Misc/Misc.fsproj
+++ b/Examples/Misc/Misc.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -40,6 +39,7 @@
     <DocumentationFile>bin\Release\Examples.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
@@ -73,7 +73,6 @@
     </ProjectReference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Examples/ToDo.UI/ToDo.UI.csproj
+++ b/Examples/ToDo.UI/ToDo.UI.csproj
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
   <PropertyGroup>
     <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
@@ -22,7 +22,6 @@
     <OutputPath>bin\Debug\</OutputPath>
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
     <PlatformTarget>AnyCPU</PlatformTarget>
@@ -31,7 +30,6 @@
     <OutputPath>bin\Release\</OutputPath>
     <DefineConstants>TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <PropertyGroup>
     <StartupObject />

--- a/Examples/ToDo.sln
+++ b/Examples/ToDo.sln
@@ -1,0 +1,23 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 2012
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "ToDo", "ToDo\ToDo.fsproj", "{4F8448C1-4B30-4708-B378-02696D6C1962}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ToDo.UI", "ToDo.UI\ToDo.UI.csproj", "{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Release|Any CPU = Release|Any CPU
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{4F8448C1-4B30-4708-B378-02696D6C1962}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{4F8448C1-4B30-4708-B378-02696D6C1962}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{4F8448C1-4B30-4708-B378-02696D6C1962}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{4F8448C1-4B30-4708-B378-02696D6C1962}.Release|Any CPU.Build.0 = Release|Any CPU
+		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Release|Any CPU.Build.0 = Release|Any CPU
+	EndGlobalSection
+EndGlobal

--- a/Hopac.sln
+++ b/Hopac.sln
@@ -1,55 +1,55 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
+# Visual Studio 2012
 VisualStudioVersion = 12.0.31101.0
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Benchmarks", "Benchmarks", "{0F31BE63-CB73-4207-94E8-158D3077F24A}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "ThreadRing", "Benchmarks\ThreadRing\ThreadRing.fsproj", "{36759B0E-103D-4985-815C-2ACFE9568334}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "ThreadRing", "Benchmarks\ThreadRing\ThreadRing.fsproj", "{36759B0E-103D-4985-815C-2ACFE9568334}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Fibonacci", "Benchmarks\Fibonacci\Fibonacci.fsproj", "{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Fibonacci", "Benchmarks\Fibonacci\Fibonacci.fsproj", "{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Chameneos", "Benchmarks\Chameneos\Chameneos.fsproj", "{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Chameneos", "Benchmarks\Chameneos\Chameneos.fsproj", "{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "PrimesStream", "Benchmarks\PrimesStream\PrimesStream.fsproj", "{C8993386-1AAD-4E21-B301-3003686B5BCC}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "PrimesStream", "Benchmarks\PrimesStream\PrimesStream.fsproj", "{C8993386-1AAD-4E21-B301-3003686B5BCC}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "AsyncOverhead", "Benchmarks\AsyncOverhead\AsyncOverhead.fsproj", "{53173B8B-A6B2-49DA-95F6-BF2BD3889143}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "AsyncOverhead", "Benchmarks\AsyncOverhead\AsyncOverhead.fsproj", "{53173B8B-A6B2-49DA-95F6-BF2BD3889143}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "PingPong", "Benchmarks\PingPong\PingPong.fsproj", "{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "PingPong", "Benchmarks\PingPong\PingPong.fsproj", "{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "ReaderWriter", "Benchmarks\ReaderWriter\ReaderWriter.fsproj", "{8FA0660F-2971-4E63-A3B4-45530F8B1F02}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "ReaderWriter", "Benchmarks\ReaderWriter\ReaderWriter.fsproj", "{8FA0660F-2971-4E63-A3B4-45530F8B1F02}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "CounterActor", "Benchmarks\CounterActor\CounterActor.fsproj", "{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "CounterActor", "Benchmarks\CounterActor\CounterActor.fsproj", "{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "PostMailbox", "Benchmarks\PostMailbox\PostMailbox.fsproj", "{3824013A-7063-470D-BE07-E64F1FE88A04}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "PostMailbox", "Benchmarks\PostMailbox\PostMailbox.fsproj", "{3824013A-7063-470D-BE07-E64F1FE88A04}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "CmlLCH", "Benchmarks\CmlLCH\CmlLCH.fsproj", "{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "CmlLCH", "Benchmarks\CmlLCH\CmlLCH.fsproj", "{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Cell", "Benchmarks\Cell\Cell.fsproj", "{68C52C21-42A4-445B-812D-1392515E2C6E}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Cell", "Benchmarks\Cell\Cell.fsproj", "{68C52C21-42A4-445B-812D-1392515E2C6E}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "StartRing", "Benchmarks\StartRing\StartRing.fsproj", "{975A9D84-8D11-437C-B247-48AD3DD3B129}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "StartRing", "Benchmarks\StartRing\StartRing.fsproj", "{975A9D84-8D11-437C-B247-48AD3DD3B129}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Streams", "Benchmarks\Streams\Streams.fsproj", "{83F3235E-2033-4752-90E9-1048781FA817}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Streams", "Benchmarks\Streams\Streams.fsproj", "{83F3235E-2033-4752-90E9-1048781FA817}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Libs", "Libs", "{43B5134E-89E6-411E-A492-6FDA8BD441FB}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Hopac", "Libs\Hopac\Hopac.fsproj", "{898D8FE3-C9BE-4115-9A13-7615AF27C048}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Hopac", "Libs\Hopac\Hopac.fsproj", "{898D8FE3-C9BE-4115-9A13-7615AF27C048}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Hopac.Extra", "Libs\Hopac.Extra\Hopac.Extra.fsproj", "{D5780B41-1F30-4057-86E8-47FC90F1FA65}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Hopac.Extra", "Libs\Hopac.Extra\Hopac.Extra.fsproj", "{D5780B41-1F30-4057-86E8-47FC90F1FA65}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Hopac.Core", "Libs\Hopac.Core\Hopac.Core.csproj", "{92AC6198-FA63-4458-AFEC-2BFBB2B0D914}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Hopac.Experimental", "Libs\Hopac.Experimental\Hopac.Experimental.fsproj", "{939C22FA-DFB0-491A-B080-F740F7FF2A8C}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Hopac.Experimental", "Libs\Hopac.Experimental\Hopac.Experimental.fsproj", "{939C22FA-DFB0-491A-B080-F740F7FF2A8C}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Hopac.Platform.Net", "Libs\Hopac.Platform.Net\Hopac.Platform.Net.fsproj", "{B285AEB6-FC96-4212-B7A2-6B73597CE607}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Hopac.Platform.Net", "Libs\Hopac.Platform.Net\Hopac.Platform.Net.fsproj", "{B285AEB6-FC96-4212-B7A2-6B73597CE607}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Examples", "Examples", "{24B2ECA5-6A25-48E3-942E-D7559D4F8C3B}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Misc", "Examples\Misc\Misc.fsproj", "{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "Misc", "Examples\Misc\Misc.fsproj", "{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Tests", "Tests", "{43B5134E-89E6-411E-A492-6FDA8BD441F9}"
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "AdHocTests", "Tests\AdHocTests\AdHocTests.fsproj", "{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}"
+Project("{f2a71f9b-5d33-465a-a702-920d77279786}") = "AdHocTests", "Tests\AdHocTests\AdHocTests.fsproj", "{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Docs", "Docs", "{6B539F10-FF76-4EA2-9A2F-594F377CEEC2}"
 	ProjectSection(SolutionItems) = preProject
@@ -78,64 +78,44 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".nuget", ".nuget", "{C2B858
 		.nuget\packages.config = .nuget\packages.config
 	EndProjectSection
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ToDo.UI", "Examples\ToDo.UI\ToDo.UI.csproj", "{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}"
-EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "ToDo", "Examples\ToDo\ToDo.fsproj", "{4F8448C1-4B30-4708-B378-02696D6C1962}"
-EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
 		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{36759B0E-103D-4985-815C-2ACFE9568334}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{36759B0E-103D-4985-815C-2ACFE9568334}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{36759B0E-103D-4985-815C-2ACFE9568334}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{36759B0E-103D-4985-815C-2ACFE9568334}.Release|Any CPU.Build.0 = Release|Any CPU
-		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Release|Any CPU.Build.0 = Release|Any CPU
-		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Release|Any CPU.Build.0 = Release|Any CPU
-		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Release|Any CPU.Build.0 = Release|Any CPU
-		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Release|Any CPU.Build.0 = Release|Any CPU
-		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Release|Any CPU.Build.0 = Release|Any CPU
-		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Release|Any CPU.Build.0 = Release|Any CPU
 		{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{0E841EC9-AE81-45CF-A404-E719BD2FBFFC}.Release|Any CPU.Build.0 = Release|Any CPU
+		{36759B0E-103D-4985-815C-2ACFE9568334}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{36759B0E-103D-4985-815C-2ACFE9568334}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{36759B0E-103D-4985-815C-2ACFE9568334}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{36759B0E-103D-4985-815C-2ACFE9568334}.Release|Any CPU.Build.0 = Release|Any CPU
 		{3824013A-7063-470D-BE07-E64F1FE88A04}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{3824013A-7063-470D-BE07-E64F1FE88A04}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{3824013A-7063-470D-BE07-E64F1FE88A04}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{3824013A-7063-470D-BE07-E64F1FE88A04}.Release|Any CPU.Build.0 = Release|Any CPU
-		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Release|Any CPU.Build.0 = Release|Any CPU
+		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Release|Any CPU.Build.0 = Release|Any CPU
+		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{53173B8B-A6B2-49DA-95F6-BF2BD3889143}.Release|Any CPU.Build.0 = Release|Any CPU
+		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{64E64C4F-5FE9-4E5E-97AC-A4F10CE7B30E}.Release|Any CPU.Build.0 = Release|Any CPU
 		{68C52C21-42A4-445B-812D-1392515E2C6E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{68C52C21-42A4-445B-812D-1392515E2C6E}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{68C52C21-42A4-445B-812D-1392515E2C6E}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{68C52C21-42A4-445B-812D-1392515E2C6E}.Release|Any CPU.Build.0 = Release|Any CPU
-		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Release|Any CPU.Build.0 = Release|Any CPU
+		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Release|Any CPU.Build.0 = Release|Any CPU
 		{83F3235E-2033-4752-90E9-1048781FA817}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{83F3235E-2033-4752-90E9-1048781FA817}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{83F3235E-2033-4752-90E9-1048781FA817}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -144,10 +124,10 @@ Global
 		{898D8FE3-C9BE-4115-9A13-7615AF27C048}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{898D8FE3-C9BE-4115-9A13-7615AF27C048}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{898D8FE3-C9BE-4115-9A13-7615AF27C048}.Release|Any CPU.Build.0 = Release|Any CPU
-		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Release|Any CPU.Build.0 = Release|Any CPU
+		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{8FA0660F-2971-4E63-A3B4-45530F8B1F02}.Release|Any CPU.Build.0 = Release|Any CPU
 		{92AC6198-FA63-4458-AFEC-2BFBB2B0D914}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{92AC6198-FA63-4458-AFEC-2BFBB2B0D914}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{92AC6198-FA63-4458-AFEC-2BFBB2B0D914}.Release|Any CPU.ActiveCfg = Release|Any CPU
@@ -156,29 +136,34 @@ Global
 		{939C22FA-DFB0-491A-B080-F740F7FF2A8C}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{939C22FA-DFB0-491A-B080-F740F7FF2A8C}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{939C22FA-DFB0-491A-B080-F740F7FF2A8C}.Release|Any CPU.Build.0 = Release|Any CPU
+		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{975A9D84-8D11-437C-B247-48AD3DD3B129}.Release|Any CPU.Build.0 = Release|Any CPU
+		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{AD67E0DD-7FFC-4630-9FA4-8C6E1CB000CA}.Release|Any CPU.Build.0 = Release|Any CPU
 		{B285AEB6-FC96-4212-B7A2-6B73597CE607}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{B285AEB6-FC96-4212-B7A2-6B73597CE607}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{B285AEB6-FC96-4212-B7A2-6B73597CE607}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{B285AEB6-FC96-4212-B7A2-6B73597CE607}.Release|Any CPU.Build.0 = Release|Any CPU
-		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314}.Release|Any CPU.Build.0 = Release|Any CPU
-		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1}.Release|Any CPU.Build.0 = Release|Any CPU
-		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08}.Release|Any CPU.Build.0 = Release|Any CPU
-		{4F8448C1-4B30-4708-B378-02696D6C1962}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{4F8448C1-4B30-4708-B378-02696D6C1962}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{4F8448C1-4B30-4708-B378-02696D6C1962}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{4F8448C1-4B30-4708-B378-02696D6C1962}.Release|Any CPU.Build.0 = Release|Any CPU
-	EndGlobalSection
-	GlobalSection(SolutionProperties) = preSolution
-		HideSolutionNode = FALSE
+		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{C8993386-1AAD-4E21-B301-3003686B5BCC}.Release|Any CPU.Build.0 = Release|Any CPU
+		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{CBE8EB36-3F88-42D3-AB00-931C70EEF81F}.Release|Any CPU.Build.0 = Release|Any CPU
+		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{D5780B41-1F30-4057-86E8-47FC90F1FA65}.Release|Any CPU.Build.0 = Release|Any CPU
+		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{EFFC7C11-2D7F-4F8C-A20A-1085F8F4184E}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(NestedProjects) = preSolution
 		{36759B0E-103D-4985-815C-2ACFE9568334} = {0F31BE63-CB73-4207-94E8-158D3077F24A}
@@ -201,7 +186,8 @@ Global
 		{B285AEB6-FC96-4212-B7A2-6B73597CE607} = {43B5134E-89E6-411E-A492-6FDA8BD441FB}
 		{7A2F0650-E5F6-4DFB-975D-C9F8B10FF314} = {24B2ECA5-6A25-48E3-942E-D7559D4F8C3B}
 		{4E4BE430-1137-43D1-97EF-206BD6ACB2C1} = {43B5134E-89E6-411E-A492-6FDA8BD441F9}
-		{E73B8A78-99C2-4E4F-B365-C1E240E0EF08} = {24B2ECA5-6A25-48E3-942E-D7559D4F8C3B}
-		{4F8448C1-4B30-4708-B378-02696D6C1962} = {24B2ECA5-6A25-48E3-942E-D7559D4F8C3B}
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
 	EndGlobalSection
 EndGlobal

--- a/Libs/Hopac.Core/Hopac.Core.csproj
+++ b/Libs/Hopac.Core/Hopac.Core.csproj
@@ -41,11 +41,10 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <Private>True</Private>
+    <Reference Include="System" />
+    <Reference Include="FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
     </Reference>
-    <Reference Include="System" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Assembly.cs" />

--- a/Libs/Hopac.Experimental/Hopac.Experimental.fsproj
+++ b/Libs/Hopac.Experimental/Hopac.Experimental.fsproj
@@ -60,7 +60,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Libs/Hopac.Extra/Hopac.Extra.fsproj
+++ b/Libs/Hopac.Extra/Hopac.Extra.fsproj
@@ -65,7 +65,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Libs/Hopac.Platform.Net/Hopac.Platform.Net.fsproj
+++ b/Libs/Hopac.Platform.Net/Hopac.Platform.Net.fsproj
@@ -16,7 +16,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>true</Optimize>
     <Tailcalls>true</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -32,6 +31,7 @@
     <DefineConstants>TRACE</DefineConstants>
     <WarningLevel>3</WarningLevel>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -64,7 +64,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Libs/Hopac/Hopac.fsproj
+++ b/Libs/Hopac/Hopac.fsproj
@@ -60,7 +60,6 @@
   <ItemGroup>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

--- a/Tests/AdHocTests/AdHocTests.fsproj
+++ b/Tests/AdHocTests/AdHocTests.fsproj
@@ -18,7 +18,6 @@
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <Tailcalls>false</Tailcalls>
     <OutputPath>bin\Debug\</OutputPath>
@@ -40,6 +39,7 @@
     <DocumentationFile>bin\Release\AdHocTests.XML</DocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <OtherFlags>--warnon:1182</OtherFlags>
+    <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
   <PropertyGroup>
     <MinimumVisualStudioVersion Condition="'$(MinimumVisualStudioVersion)' == ''">11</MinimumVisualStudioVersion>
@@ -73,8 +73,7 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Libs\Hopac.Platform.Net\Hopac.Platform.Net.fsproj">
       <Name>Hopac.Platform.Net</Name>
-      <Project>{b285aeb6-fc96-4212-b7a2-6b73597ce607}</Project>
-      <Private>True</Private>
+      <Project>{B285AEB6-FC96-4212-B7A2-6B73597CE607}</Project>
     </ProjectReference>
     <ProjectReference Include="..\..\Libs\Hopac\Hopac.fsproj">
       <Name>Hopac</Name>
@@ -82,11 +81,9 @@
     </ProjectReference>
     <Reference Include="FsCheck">
       <HintPath>..\..\packages\FsCheck.1.0.4\lib\net45\FsCheck.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="FSharp.Core">
       <HintPath>..\..\packages\FSharp.Core.3.1.2.1\lib\net40\FSharp.Core.dll</HintPath>
-      <Private>True</Private>
     </Reference>
     <Reference Include="mscorlib" />
     <Reference Include="System" />

